src/rust/engine/grpc_util/src/tls.rs

```diff
@@ -48,6 +48,47 @@ impl Config {
             certificate_check: CertificateCheck::Enabled,
         })
     }
+
+    pub fn new_from_files(
+        root_ca_cert_file: &Option<PathBuf>,
```

Comment:
Why not `Option<&Path>` (which `PathBuf` dereferences to via `&path`). (Given `&Path` is to `PathBuf` as `&str` is to `String`.)

Reply:
You would probably use `opt.as_ref()` at the call site.

Reply:
Ah yes, good catch, these don't need to be PathBuf, even though they likely always will be. 

Fixed (with `opt.as_deref()`, since we have to `deref` the contained object).